examples/direct_fidelity_estimation.py

```diff
@@ -52,65 +50,129 @@ def compute_characteristic_function(circuit: cirq.Circuit,
     return trace, prob
 
 
+async def estimate_characteristic_function(
+        circuit: cirq.Circuit, P_i: Tuple[cirq.Gate, ...],
+        qubits: List[cirq.Qid], simulator: cirq.DensityMatrixSimulator,
+        samples_per_term: int):
+    n_qubits = len(P_i)
+
+    pauli_string = cirq.PauliString(dict(zip(qubits, P_i)))
+    qubit_map = dict(zip(qubits, range(n_qubits)))
+
+    p = cirq.PauliSumCollector(circuit=circuit,
+                               observable=pauli_string,
+                               samples_per_term=samples_per_term)
+
+    completion = p.collect_async(sampler=simulator)
+    assert await completion is None
+
+    sigma_i = p.estimated_energy()
+    assert np.isclose(sigma_i.imag, 0.0, atol=1e-6)
+    sigma_i = sigma_i.real
+
+    return sigma_i
+
+
 def direct_fidelity_estimation(circuit: cirq.Circuit, qubits: List[cirq.Qid],
-                               noise: cirq.NoiseModel, n_trials: int):
+                               noise: cirq.NoiseModel, n_trials: int,
+                               samples_per_term: int):
```

Comment:
Add a docstring. Especially note that setting `samples_per_term` to `0` changes the behavior of this function

Reply:
Done.